bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityType.java

```diff
@@ -457,6 +457,11 @@ public String toFullString() {
         } else if (target == PercentType.class) {
             if (Units.PERCENT.equals(getUnit())) {
                 return target.cast(new PercentType(toBigDecimal()));
+            } else {
+                QuantityType<T> inPercent = toUnit(Units.PERCENT);
+                if (inPercent != null) {
+                    return inPercent.as(target);
+                }
             }
```

Comment:
I think you should check if the dimension is "dimensionless" and return `null` if that is not the case.

Reply:
Sure, agree it would sense that non-compatible units (example meter) are disregarded. It would be even more breaking change but then again how many are relying such odd logic

How to check dimension specifically? I have been using the unit conversion to support transformed units (%/10) as well.